File: HNLineAnimationTextField-OC/HNLineAnimationTextFieldDemo-OC/HNLineShapeLayer.m
//
//  HNLineShapeLayer.m
//  HNLineAnimationTextFieldDemo-OC
//
//  Created by zakariyyaSv on 16/4/24.
//  Copyright © 2016年 Zakariyya. All rights reserved.
//

#import "HNLineShapeLayer.h"
#import <UIKit/UIKit.h>

@implementation HNLineShapeLayer

- (instancetype)init{
    if (self = [super init]) {
        self.fillColor = [UIColor clearColor].CGColor;
        self.lineJoin = kCALineJoinRound;
        self.lineCap = kCALineCapRound;
        self.opaque = YES;
    }
    return self;
}

@end

File: HNLineAnimationTextFieldDemo-swift/HNLineAnimationTextFieldDemo-swift/HNLineAnimationTextField/HNLineShapeLayer.swift
//
//  HNLineShapeLayer.swift
//  HNLineAnimationTextFieldDemo-swift
//
//  Created by zakariyyaSv on 16/4/26.
//  Copyright © 2016年 Zakariyya. All rights reserved.
//

import UIKit

class HNLineShapeLayer: CAShapeLayer {
    override init() {
        super.init()
        fillColor = UIColor.clearColor().CGColor
        lineJoin = kCALineJoinRound
        lineCap = kCALineCapRound
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

}
